qkernel/src/lib.rs

```diff
@@ -511,82 +572,90 @@ cfg_if::cfg_if! {
                 let pd_adr = PAGE_MGR_HOLDER.Addr();
                 HyperCall64_init(qlib::HYPERCALL_SHARESPACE_INIT, shared_space as u64, pd_adr, 0, 0);
 
-                
-                SHARESPACE.SetValue(shared_space as u64);                
 
+                SHARESPACE.SetValue(shared_space as u64);
 
-                SingletonInit();
 
+                SingletonInit();
+                LOG_AVAILABLE.store(true, Ordering::Release);
                 SetVCPCount(vcpuCnt as usize);
                 VCPU_ALLOCATOR.Print();
                 VCPU_ALLOCATOR.Initializated();
-        
+
                 InitTsc();
                 InitTimeKeeper(vdsoParamAddr);
                 {
                     let kpt = &KERNEL_PAGETABLE;
-        
+
                     let vsyscallPages = PAGE_MGR.VsyscallPages();
                     kpt.InitVsyscall(vsyscallPages);
                 }
-        
+
                 GlobalIOMgr().InitPollHostEpoll(SHARESPACE.HostHostEpollfd());
-        
+
                 VDSO.Initialization(vdsoParamAddr);
-        
+
                 // release other vcpus
                 HyperCall64(qlib::HYPERCALL_RELEASE_VCPU, 0, 0, 0, 0);
             } else {
+                x86_64::instructions::tlb::flush_all();
+                interrupt::init();
                 InitGs(id);
+                if IS_SEV_SNP.load(Ordering::Acquire){
+                    InitGhcb(id as usize);
+                }
                 //PerfGoto(PerfType::Kernel);
             }
-        
+
             SHARESPACE.IncrVcpuSearching();
             taskMgr::AddNewCpu();
             RegisterSysCall(syscall_entry as u64);
-        
-            //interrupts::init_idt();
-            interrupt::init();
-        
+
+            //Different from normal vm, mxcsr will not be set by kvm, should set it mannualy
+            if IS_SEV_SNP.load(Ordering::Acquire){
+                let mxcsr_value = 0x1f80;
```

Comment:
Could you add a short comment to explain what the mask means... for readability.

Reply:
actually all these masks and hardcoded values should be constant defs.